src/pudl/extract/epacems.py

```diff
@@ -172,71 +122,18 @@ def __init__(self, datastore: Datastore):
         """Construct datastore wrapper for loading raw EPA CEMS data into dataframes."""
         self.datastore = datastore
 
-    def get_data_frame(self, partition: EpaCemsPartition) -> pd.DataFrame:
+    def get_data_frame(self, partition: EpaCemsPartition) -> pl.LazyFrame:
         """Constructs dataframe from a zipfile for a given (year_quarter) partition."""
         with (
             self.datastore.get_zipfile_resource(
                 "epacems", **partition.get_filters()
             ) as zf,
             zf.open(str(partition.get_quarterly_file()), "r") as csv_file,
         ):
-            df = self._csv_to_dataframe(
-                csv_file,
-                ignore_cols=API_IGNORE_COLS,
-                rename_dict=API_RENAME_DICT,
-                dtype_dict=API_DTYPE_DICT,
+            lf = pl.scan_csv(csv_file, schema_overrides=API_DTYPE_DICT)
+            lf = (
+                lf.drop(list(set(lf.columns) - set(API_RENAME_DICT.keys())))
+                .cast(API_DTYPE_DICT, strict=False)
+                .rename(API_RENAME_DICT, strict=False)
```

Comment:
So concise!

Reply:
At least for simple things it feels really clean to me too. There's some stuff with using "structs" to deal with multi-column conditions that would use indices in pandas that feels kinda jank, but maybe I'm just not familiar enough with it.